stdlib/jwt/src/main/ballerina/jwt/jwt_issuer.bal

```diff
@@ -122,16 +121,10 @@ function buildPayloadString(JwtPayload payload, boolean audienceAsArray) returns
         payloadJson[JTI] = jti;
     }
     var aud = payload["aud"];
-    if (aud is string[] && aud.length() > 0) {
-        if (audienceAsArray) {
-            payloadJson[AUD] = aud;
-        } else {
-            if (aud.length() == 1) {
-                payloadJson[AUD] = aud[0];
-            } else {
-                payloadJson[AUD] = aud;
-            }
-        }
+    if (aud is string) {
+        payloadJson[AUD] = aud;
+    } else if (aud is string[] && aud.length() > 0) {
```

Comment:
IMO, we don't have to control this. If user providers an empty array, we send out an empty array. IMO, that will give fully flexibility to the user. Any problem with doing so?
```suggestion
    } else if (aud is string[]) {
```

Reply:
Yeah. Agree on this. Otherwise, we are over controlling the flexibility given by the spec. I will fix that.

Reply:
Fixed with https://github.com/ballerina-platform/ballerina-lang/pull/16174/commits/08ea771706864b1683f4dd60a1998a36159d1679